### PR TITLE
Lint markdown + CI

### PR DESCRIPTION
**Pull request by weswigham**
This is mostly for bikeshedding on what rules should be on - right now we fail a lot, like trailing whitespace, no hard tabs, and so on.

### CODE REVIEW

File: lint.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+var markdownlint = require("markdownlint");
+var glob = require("glob");
+
+var options = {
+  files: glob.sync("**/*.md", {ignore: "node_modules/**/*"}),
+  config: {
+      MD013: false // Line length
```
Comment (DanielRosenwasser) [inline lint.js]: Line length of what?

Comment (DanielRosenwasser) [inline lint.js]: I'd consider 100 or 120.

Comment (weswigham) [inline lint.js]: Why not just leave it on false and abide by the every-line-a-sentence rule?

**Comment by DanielRosenwasser** [comment]
MD004 | Unordered list style | "asterisk"
MD005 | Inconsistent indentation for list items at the same level |
MD006 | Consider starting bulleted lists at the beginning of the line |
MD007 | Unordered list indentation | Figure out how we already use this; if we don't, 4.
MD009 | Trailing spaces |
MD010 | Filter out hard tabs |
MD011 | Reversed link syntax |
MD018 | No space after hash on atx style header |
MD019 | Multiple spaces after hash on atx style header |
MD022 | Headers should be surrounded by blank lines |
MD023 | Headers must start at the beginning of the line |


Eventually I'd like to have `MD024 - Multiple headers with the same content`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "name": "typescript-handbook",
+  "version": "1.0.0",
+  "description": "The TypeScript Handbook is a comprehensive guide to the TypeScript language",
+  "main": "none",
+  "scripts": {
+    "test": "node lint.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/weswigham/TypeScript-Handbook.git"
```
Comment (sandersn) [inline package.json]: Shouldn't it be Microsoft/Typescript-Handbook?

Comment (weswigham) [inline package.json]: aye, this is just the default output of npm init so I could write in the test command, lemme fix it up

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "typescript-handbook",
+  "version": "1.0.0",
```
Comment (DanielRosenwasser) [inline package.json]: 1.6.0

File: package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Microsoft/TypeScript-Handbook.git"
+  },
+  "author": "",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/Microsoft/TypeScript-Handbook/issues"
+  },
+  "homepage": "https://github.com/Microsoft/TypeScript-Handbook#readme",
```
Comment (DanielRosenwasser) [inline package.json]: We're going to need to change this down the road I guess

**Comment by DanielRosenwasser** [comment]
:+1:

**Pull request merged**